crates/node-primitives/src/lib.rs

```diff
@@ -57,5 +57,7 @@ pub enum MutateCallError {
         application_id: calimero_primitives::application::ApplicationId,
     },
     NoConnectedPeers,
+    TransactionRejected,
```

Comment:
Y'know, thinking about it.. rejected transactions aren't the client's fault, the node should be able to resubmit with new knowledge of the current transaction hash